contracts/scripts/ci/generate_slither_report.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+function check_chainlink_dir() {
+  local param_dir="chainlink"
+  current_dir=$(pwd)
+
+  current_base=$(basename "$current_dir")
+
+  if [ "$current_base" != "$param_dir" ]; then
+    >&2 echo "The script must be run from the root of $param_dir directory"
+    exit 1
+  fi
+}
+
+check_chainlink_dir
+
+if [ "$#" -lt 5 ]; then
+  >&2 echo "Generates Markdown Slither reports and saves them to a target directory."
+  >&2 echo "Usage: $0 <https://github.com/ORG/REPO/blob/COMMIT/> <config-file> <root-directory-with–contracts> <comma-separated list of contracts> <where-to-save-reports> [slither extra params]"
+exit 1
+fi
+
+REPO_URL=$1
+CONFIG_FILE=$2
+SOURCE_DIR=$3
+FILES=${4// /}  # Remove any spaces from the list of files
+TARGET_DIR=$5
+SLITHER_EXTRA_PARAMS=$6
+
+run_slither() {
+    local FILE=$1
+    local TARGET_DIR=$2
+
+    # needed, because the action we use returns all modified files, also deleted ones and we must skip those
+    if [ ! -f "$FILE" ]; then
+      echo "Warning: File not found: $FILE"
+      echo "Skipping..."
+      return
+    fi
+
+    source ./contracts/scripts/ci/select_solc_version.sh "$FILE"
+    if [ $? -ne 0 ]; then
+        >&2 echo "Error: Failed to select Solc version for $FILE"
+        exit 1
+    fi
+
+    SLITHER_OUTPUT_FILE="$TARGET_DIR/$(basename "${FILE%.sol}")-slither-report.md"
+
+    output=$(slither --config-file "$CONFIG_FILE" "$FILE" --checklist --markdown-root "$REPO_URL" --fail-none $SLITHER_EXTRA_PARAMS)
```

Comment:
suggestion: I am thinking we might want to fail on high impact findings here and if it is a false positive then we update the config file to exclude the detector? one downside I see is if a detector should be ignored only on some files. cc @RensR

Reply:
up to you guys, my idea was to make it purely informative for now, but you know best what you need :-)

Reply:
Maybe nice to follow up, but this is fine for now